components/infobox/commons/infobox_league.lua

```diff
@@ -0,0 +1,284 @@
+local Class = require('Module:Class')
+local Cell = require('Module:Infobox/Cell')
+local Infobox = require('Module:Infobox')
+local Template = require('Module:Template')
+local Table = require('Module:Table')
+local Namespace = require('Module:Namespace')
+local String = require('Module:String')
+local Links = require('Module:Links')
+local Flags = require('Module:Flags')
+local Localisation = require('Module:Localisation')
+
+local getArgs = require('Module:Arguments').getArgs
+
+local League = Class.new()
+
+function League.run(frame)
+    return League:createInfobox(frame)
+end
+
+function League:createInfobox(frame)
+    local args = getArgs(frame)
+    self.frame = frame
+    self.pagename = mw.title.getCurrentTitle().title
+    self.name = args.name or self.pagename
+
+    if args.wiki == nil then
+        return error('Please provide a wiki!')
+    end
+
+    local infobox = Infobox:create(frame, args.wiki)
+
+    infobox :name(args.name)
+            :image(args.image, args.default)
+            :centeredCell(args.caption)
+            :header('League Information', true)
+            :fcell(Cell :new('Series')
+                        :options({})
+                        :content(
+                            League:_createSeries(frame, args.series, args.abbrevation),
+                            League:_createSeries(frame, args.series2, args.abbrevation2)
+                        )
+                        :make()
+            )
+            :fcell(Cell  :new('Organizer')
+                        :options({})
+                        :content(
+                            unpack(League:_createOrganizers(args))
+                        )
+                        :make()
+            )
+            :cell('Sponsor(s)', args.sponsor)
+            :fcell(Cell  :new('Mode')
+                        :options({})
+                        :content(args.mode)
+                        :variables({key = 'mode', value = args.mode})
+                        :make()
+            )
```

Comment:
should make this a function because wikis use quite different stuff here
(some have switches implemented that only allow certain modes, some use "gamemode" as argument)

we also should add a cell for Version/Game Version (only 3 main wikis do not use such a display), should also be a function the wikis can overwrite

![Screenshot 2021-08-17 14 37 22](https://user-images.githubusercontent.com/75081997/129736177-a6264d9c-9bad-456e-9f76-77ef857569ca.png)

Reply:
Should mode just be wiki specific then?

Reply:
Leaving it out, the wikis can solve it for themselves